controllers/apps/transformer_cluster_status.go

```diff
@@ -173,11 +173,14 @@ func (t *clusterStatusTransformer) removeInvalidCompStatus(transCtx *clusterTran
 }
 
 // removeInnerCompStatus removes the component of status.components which is created by simplified API.
-func (t *clusterStatusTransformer) removeInnerCompStatus(cluster *appsv1alpha1.Cluster) {
+func (t *clusterStatusTransformer) removeInnerCompStatus(transCtx *clusterTransformContext, cluster *appsv1alpha1.Cluster) {
 	compSpecs := make([]*appsv1alpha1.ClusterComponentSpec, 0)
 	for i := range cluster.Spec.ComponentSpecs {
 		compSpecs = append(compSpecs, &cluster.Spec.ComponentSpecs[i])
 	}
+	for _, v := range transCtx.ShardingComponentSpecs {
+		compSpecs = append(compSpecs, v...)
```

Comment:
it is recommended not to list components generated by sharding in the cluster status

Reply:
> it is recommended not to list components generated by sharding in the cluster status

The sharding components' status display has been temporarily removed and will need to be redesigned later on.

Reply:
> > it is recommended not to list components generated by sharding in the cluster status